docs/src/turing.md

```diff
@@ -0,0 +1,29 @@
+## Using AdvancedHMC with Turing
```

Comment:
I'm a bit uncertain if we want this here; it seems more suitable to be put in Turing.jl, no?

And, if we keep it here, we should at least make the code blocks below `@example turing` or something so that they're run, given that the `externalsampler` is quite a new features and thus prone to breaking changes.

Reply:
Done!